src/compiler/scala/tools/nsc/transform/patmat/MatchTreeMaking.scala

```diff
@@ -429,7 +429,10 @@ trait MatchTreeMaking extends MatchCodeGen with Debugging {
                 || definedInStaticLocation(expectedTp)
                 || testedBinder.info <:< expectedTp
                 || prefixAligns) orig
-              else gen.mkAttributedQualifierIfPossible(expectedPrefix) match {
+              else (expectedPrefix match {
+                case RefinedType(parents, _) if !parents.exists(_.isStable) => None
```

Comment:
can this (or an equivalent) test be part of `prefixAligns`?

```
                          pre.termSymbol match {
                            case NoSymbol => false
```

Reply:
I don't think so because IIRC in the test case it's not the expected type's prefix, but the computed expectedPrefix which is a refined type.